lib/constable_web/controllers/api/announcement_controller.ex

```diff
@@ -8,8 +8,44 @@ defmodule ConstableWeb.Api.AnnouncementController do
 
   plug(Constable.Plugs.Deslugifier, slugified_key: "id")
 
-  def index(conn, _params) do
-    announcements = Repo.all(Announcement)
+  def index(conn, params = %{"after" => after_id}) do
+    limit = Map.get(params, "page_size", 50)
+    cursor = Repo.get(Announcement, after_id)
+
+    announcements =
+      Announcement
+      |> where([a], a.inserted_at < ^cursor.inserted_at)
+      |> order_by(desc: :inserted_at)
+      |> limit(^limit)
+      |> Repo.all()
+
+    render(conn, "index.json", announcements: announcements)
+  end
+
+  def index(conn, params = %{"before" => before_id}) do
+    limit = Map.get(params, "page_size", 50)
+    cursor = Repo.get(Announcement, before_id)
+
+    announcements =
+      Announcement
+      |> where([a], ^cursor.inserted_at < a.inserted_at)
+      |> order_by(:inserted_at)
+      |> limit(^limit)
+      |> Repo.all()
+      |> Enum.reverse()
```

Comment:
Curious, why are we doing an `Enum.reverse()` here instead of sorting the articles in sql?

Reply:
AFAIK, a `LIMIT` clause doesn't work in the reverse direction in SQL. So in order to get the 50 articles _behind_ a given article, we came up with this method:

1. Reverse the sort order in SQL and apply the limit.
2. Fetch the records into memory and reverse them again to return to the original order (descending by publish time).

Does that make sense?

Reply:
👍 that does make sense. Thanks for the explanation!